parlai/core/image_featurizers.py

```diff
@@ -57,13 +57,35 @@ def init_cnn(self, opt):
             print('[ Using CUDA ]')
             torch.cuda.set_device(opt.get('gpu', -1))
 
-        cnn_type, layer_num = self._image_mode_switcher()
-
-        # initialize the pretrained CNN using pytorch.
-        CNN = getattr(torchvision.models, cnn_type)
-
-        # cut off the additional layer.
-        self.netCNN = nn.Sequential(*list(CNN(pretrained=True).children())[:layer_num])
+        if 'resnet' in self.image_mode:
+            cnn_type, layer_num = self._image_mode_switcher()
+            # initialize the pretrained CNN using pytorch.
+            CNN = getattr(torchvision.models, cnn_type)
+
+            # cut off the additional layer.
+            self.netCNN = nn.Sequential(
+                *list(CNN(pretrained=True).children())[:layer_num])
+        elif 'resnext' in self.image_mode:
+            try:
+                model = torch.hub.load('facebookresearch/WSL-Images', self.image_mode)
+                self.netCNN = nn.Sequential(
+                    *list(model.children())[:-1])
```

Comment:
leave a comment saying you're cutting off the classification layer?

Reply:
good idea